### PR TITLE
fix: record the completed length of the last calculation as the transison length

### LINKED ISSUE #42958: Typography.Paragraph刚好出现滚动条时一直闪屏
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/s7um3s?file=/demo.tsx)

### Steps to reproduce

1.此为antd官网demo,将ellipsis设置成3行
2.拖动浏览器边框使demo刚好出现滚动条时，会一直闪
3.在出现滚动条的边界反复试一下

### What is expected?

出现正常的滚动条，文本没有屏闪现象

### What is actually happening?

文本屏闪

| Environment | Info |
| --- | --- |
| antd | 4.24.10 |
| React | react17 |
| System | windows10 |
| Browser |  114.0.5735.110（正式版本） |

### PR DESCRIPTION
**Pull request by bbb169**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/42958

### 💡 Background and solution

The bug is caused by below codes:
if (!enabledMeasure || walkingState !== DONE_WITH_ELLIPSIS) {
      return children(nodeList, false);
}

when ellipsis is walking, will use all the nodelist as temporary text, this caused space is not enough, so page scroll bar will show up
solution is record the completed length of the last calculation as the transition length in the calculation
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      record the completed length of the last calculation as the transition length in the calculation     |
| 🇨🇳 Chinese |     记录上次计算完成的文字长度作为计算中的过渡文字长度      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 741a1de</samp>

Improve ellipsis feature in `Typography` component by reusing optimal text length. Refactor `Ellipsis.tsx` to use a state variable `lastLen`.

### <samp>🤖 Generated by Copilot at 741a1de</samp>

* Optimize the performance of the ellipsis component by introducing a new state variable `lastLen` to store the last length of the text that was successfully rendered with ellipsis ([link](https://github.com/ant-design/ant-design/pull/43058/files?diff=unified&w=0#diff-4db52a9940b2f4bc55f62f1d30dd05ca8889806a7525ddac72d7b0a492bae93dR89-R90), [link](https://github.com/ant-design/ant-design/pull/43058/files?diff=unified&w=0#diff-4db52a9940b2f4bc55f62f1d30dd05ca8889806a7525ddac72d7b0a492bae93dR161))
* Use the `lastLen` as a temporary width for the text if it exists, to avoid rendering the text with the full length and then shrinking it, which could cause unnecessary layout shifts and performance issues ([link](https://github.com/ant-design/ant-design/pull/43058/files?diff=unified&w=0#diff-4db52a9940b2f4bc55f62f1d30dd05ca8889806a7525ddac72d7b0a492bae93dR103-R105))

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
ci 挂了

**Comment by bbb169** [comment]
> ci 挂了
image
已改

**Comment by afc163** [comment]
@MadCcc 看看

**Review by kiner-tang — APPROVED** [review]

**Comment by zombieJ** [comment]
这个 Demo 挂了：

截屏2023-08-28 23 03 11

**Comment by bbb169** [comment]
> 这个 Demo 挂了：
> 
> 截屏2023-08-28 23 03 11

这个测试用例是错的，因为我写不通了所以传上来想看看错在哪了，不行的话我晚些时候再试试另一种方式吧

image

File: components/typography/__tests__/ellipsis.test.tsx [hunk 1]
```diff
@@ -433,4 +433,78 @@ describe('Typography.Ellipsis', () => {
     });
     mockRectSpy.mockRestore();
   });
+
+  it('should not throw default dom nodes', async () => {
+    let currentWidth = 100;
+    // string conut is different with different width
```
Comment (zombieJ) [inline components/typography/__tests__/ellipsis.test.tsx:439]: typo: count

**Review by zombieJ — APPROVED** [review]

**Pull request merged**